src/UniStaker.sol

```diff
@@ -802,7 +802,8 @@ contract UniStaker is INotifiableRewardReceiver, Multicall, EIP712, Nonces {
 
     uint256 _reward = scaledUnclaimedRewardCheckpoint[_beneficiary] / SCALE_FACTOR;
     if (_reward == 0) return;
-    scaledUnclaimedRewardCheckpoint[_beneficiary] = 0;
+    scaledUnclaimedRewardCheckpoint[_beneficiary] =
+      scaledUnclaimedRewardCheckpoint[_beneficiary] - (_reward * SCALE_FACTOR);
```

Comment:
Since what we're doing here is a bit counterintuitive, it might be nice to leave an inline comment documenting that this is to retain sub-wei dust that would be left due to the precision loss.